packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -129,6 +129,11 @@ Future<Process> runCommand(
 /// If [filter] is non-null, all lines that do not match it are removed. If
 /// [mapFunction] is present, all lines that match [filter] are also forwarded
 /// to [mapFunction] for further processing.
+///
+/// If [detachFilter] is non-null, the returned future will complete with exit code `0`
+/// when the process outputs something matching [detachFilter] to stderr. The process will
+/// continue in the background, and the final exit code will not be reported. [filter] is
+/// not considered on lines matching [detachFilter].
```

Comment:
Can we get some tests that this does what we want it to?

Reply:
Done.